### PR TITLE
Reopen #22481 Fixed modal closing issue

### PR DESCRIPTION
**Pull request by Biki-das**
@bvaughn  hope this is fine

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/views/hooks.js [hunk 1]
```diff
@@ -239,7 +239,8 @@ export function useModalDismissSignal(
       ownerDocument = ((modalRef.current: any): HTMLDivElement).ownerDocument;
       ownerDocument.addEventListener('keydown', handleDocumentKeyDown);
       if (dismissOnClickOutside) {
-        ownerDocument.addEventListener('click', handleDocumentClick);
+        ownerDocument.addEventListener('click', handleDocumentClick, true);
+        ownerDocument.removeEventListener('click', handleDocumentClick, true);
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/devtools/views/hooks.js:243]: This isn't quite right. You're adding (and then immediately removing) the listener.

Here's what you what:
```js
      // It's important to listen to the ownerDocument to support the browser extension.
      // Here we use portals to render individual tabs (e.g. Profiler),
      // and the root document might belong to a different window.
      ownerDocument = ((modalRef.current: any): HTMLDivElement).ownerDocument;
      ownerDocument.addEventListener('keydown', handleDocumentKeyDown);
      if (dismissOnClickOutside) {
        ownerDocument.addEventListener('click', handleDocumentClick, true);
      }
    }, 0);

    return () => {
      if (timeoutID !== null) {
        clearTimeout(timeoutID);
      }

      if (ownerDocument !== null) {
        ownerDocument.removeEventListener('keydown', handleDocumentKeyDown);
        ownerDocument.removeEventListener('click', handleDocumentClick, true);
      }
    };
```

File: packages/react-devtools-shared/src/devtools/views/hooks.js [hunk 2]
```diff
@@ -232,6 +232,7 @@ export function useModalDismissSignal(
     // In that case, we don't want to listen to the pre-existing event.
     let timeoutID = setTimeout(() => {
       timeoutID = null;
+      ``;
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/devtools/views/hooks.js:235]: What's this?

Comment (Biki-das) [reply packages/react-devtools-shared/src/devtools/views/hooks.js:235]: Yeah sure

**Comment by Biki-das** [comment]
@bvaughn sorry to disturb you , but i actually learned today , thanks for your patience and feedback

**Comment by bvaughn** [comment]
The code looks right now 👍🏼 but please run `yarn prettier` to fix the styling/lint CI failure. :)

**Comment by Biki-das** [comment]
> The code looks right now 👍🏼 but please run `yarn prettier` to fix the styling/lint CI failure. :)

Okay!

**Comment by Biki-das** [comment]
> The code looks right now 👍🏼 but please run `yarn prettier` to fix the styling/lint CI failure. :)

Can i run that without pulling a PR request or i need to do a fresh commit again @bvaughn

**Comment by bvaughn** [comment]
Just run it locally, then commit and push an update :)

```sh
yarn prettier
git add .
git commit -m "Formatting"
git push Biki-das localmain
```

**Comment by Biki-das** [comment]
> The code looks right now 👍🏼 but please run `yarn prettier` to fix the styling/lint CI failure. :)

Can i run that without pulling a PR request or i need to do a fresh commit again @bvaughn  thanks a lot! You made my day!

**Comment by bvaughn** [comment]
Wherever you checked out your fork of this repo to create your PR initially (I don't know where)

**Comment by bvaughn** [comment]
I just made the commit for you. Give it a sec to see if it works

**Comment by Biki-das** [comment]
> I just made the commit for you. Give it a sec to see if it works
> 

Yeah works! ✌✌

**Comment by bvaughn** [comment]
There we go.

**Comment by Biki-das** [comment]
Would you like to review 2 more PR of mine?

**Pull request merged**